node/src/reactor/joiner.rs

```diff
@@ -557,6 +557,9 @@ impl reactor::Reactor for Reactor {
                     });
                     self.dispatch_event(effect_builder, rng, event)
                 }
+                Message::FinalitySignature(_) => {
+                    unimplemented!("Finality signatures not handled in joiner reactor")
```

Comment:
But won't we receive those messages anyway? Shouldn't we ignore them instead of crash?

Reply:
First, it was captured by the "capture-it-all" clause and I wanted to make sure that we have IDE support in the future. My idea was that maybe, in the future, when downloading linear chain we will download finality signatures separately from the blocks (not now, since currently block _contains_ all signatures/proofs). BUT you just made me realize that this will cause a joining node to panic! :grimacing: I will change it.